algorithms/spot_prediction/reeke.py

```diff
@@ -274,21 +274,19 @@ def _p_limits(self):
         # The correct sign is determined by whether the plane normal vector is
         # more closely parallel or antiparallel to the beam direction.
 
-        sign = cmp(dp_beg, 0)
-
-        limits = [
-            ((sign * s * self._source.length()) + dp_beg) / p_dist
-            for s in (-1, 1)
-        ]
+        if dp_beg == 0:
```

Comment:
I'm checking the original algorithm again now. The magnitudes of `dp_beg` and `dp_end` are always >> 0, so `limits = [0,0]` will never arise. I'm noting this here for when any changes are translated over to the C++ implementation.

Reply:
that would explain why the mistake I found made no difference - both values would always have the same (positive) sign

Reply:
I meant |`dp_beg`| >> 0, but `sign` could still be {-1,1}, just never 0